dashboard/app/controllers/aichat_controller.rb

```diff
@@ -128,10 +128,16 @@ def chat_completion
       end
     end
 
-    if params[:aichatModelCustomizations] != JSON.parse(session.model_customizations) ||
-        params[:storedMessages] != JSON.parse(session.messages)
+    if params[:aichatModelCustomizations] != JSON.parse(session.model_customizations)
       return false
     end
+    
+    # Compare stored messages excluding timestamps
+    sessions_stored_messages = JSON.parse(session.messages).map { |message| message.except('timestamp') }
```

Comment:
Thanks for doing this! Could we be explicit about the keys we want to include (rather than excluding)? Thinking something like:

`JSON.parse(session.messages).map{ |message| message.slice('chatMessageText', 'role', 'status') }`

I can do this as a follow-up as well, but would be great to add a test that an extra key (say, a timestamp 😬 ) doesn't generate a new session.

Reply:
Sure - will update!